tools/ci-build.sh

```diff
@@ -58,6 +61,15 @@ if [ "$IN_DOCKER" == "TRUE" ]; then
   function sudo() { command "$@"; }
 fi
 
+# Resolve the matrix in the static builds to the right cmake settings.
```

Comment:
Can we move this check into the YAML? I have been hoping to simplify this build script to a point where we can use it like a normal installation script.

https://stackoverflow.com/questions/66025220/paired-values-in-github-actions-matrix